modules/webgl/src/adapter/resources/webgl-render-pass.ts

```diff
@@ -28,9 +28,22 @@ export class WEBGLRenderPass extends RenderPass {
     super(device, props);
     this.device = device;
 
+    // If no viewport is provided, apply reasonably defaults
+    let viewport;
+    if (!props?.parameters?.viewport) {
+      if (props?.framebuffer) {
+        // Set the viewport to the size of the framebuffer
+        const {width, height} = props.framebuffer;
+        viewport = [0, 0, width, height];
+      } else {
+        // Instead of using our own book-keeping, we can just read the values from the WebGL context
+        viewport = [0, 0, device.gl.drawingBufferWidth, device.gl.drawingBufferHeight];
```

Comment:
Optional:

```suggestion
        viewport = [0, 0, ...device.getSize()];
```

Reply:
> viewport = [0, 0, ...device.getSize()];

I used `device.getCanvasContext().getDrawingBufferSize()`.  I am not sure about why have a `device.getSize()` - it seems it is only available on WebGLDevice and NullDevice. I will put up a separate PR to propose removing it.